crates/codegen/language/definition/src/model/utils/mod.rs

```diff
@@ -1,5 +1,78 @@
+use crate::model::{self, Item};
+use semver::Version;
+use std::collections::BTreeSet;
+
 mod identifier;
 mod version_specifier;
 
 pub use identifier::*;
 pub use version_specifier::*;
+
+/// Collects all versions that changes the language grammar in a breaking way.
+pub fn collect_breaking_versions(lang: &model::Language) -> BTreeSet<Version> {
+    let first = lang.versions.first().cloned().unwrap();
+    let mut res = BTreeSet::from_iter([first]);
+
+    fn add_spec(res: &mut BTreeSet<Version>, spec: &Option<model::VersionSpecifier>) {
```

Comment:
since `add_spec` is not capturing anything outside it, I suggest moving it outside `collect_breaking_versions` to make it easier to read.

Reply:
This is only meant as a helper - the only valid usage is to start with a `BTreeSet::new()` as the accumulator. I'm happy to convert this to a closure that captures the `res` instead, if you'd like.

Reply:
Done in b65b136a87de3e700f1bfa7cb421a3f4a0a42f28